### ISSUE TITLE
copy raw files to our S3 bucket for archiving

### ISSUE THREAD

**Issue by infinitewarp** [project-koku/koku#1023]
## User Story
As a Koku user, I want to get all files that were used in generating my reported data.

One of the prerequisites to achieving that goal is to copy the raw files to our S3 bucket for long-time storage. Then when the user requests an export, we can reference those archived copies in our S3 bucket.

## Assumptions and Questions
- Files should be stored hierarchically with the account ID, provider type, year, month, and day in the S3 bucket using a pattern like:
  - /data_archive/acct_12345/aws/2019/08/01/filename.csv (for day-specific files)
  - /data_archive/acct_12345/aws/2019/08/00/filename.csv (for month-specific files)
- If a file with the same name already exists at the path, we will overwrite it.
- All files retrieved via `koku.masu.processor._tasks.download._get_report_files` should be saved to S3.
- See also: https://redhatmbu.aha.io/epics/COST-E-97

## Acceptance Criteria

- [ ] When the periodic task `masu.processor.tasks.get_report_files` runs, all files it downloaded are saved to S3.
- [ ] Files for different accounts, provider types, and dates are stored in different directories following the described pattern.
- [ ] The S3 bucket is configurable with values set in appropriate places for qa, ci, stage, and prod.
- [ ] The root of the path is configurable with a default value of "/data_archive/".

**Comment by infinitewarp** [comment]
Reopening because it came up in recent product discussions.

**Comment by infinitewarp** [comment]
@chargio: @jlalbertson would like to have a chat about the need/priority of Koku/Red Hat retaining a copy of the customer's original files since the customer _should_ already have access to their own files.

**Comment by chargio** [comment]
Once they are stored in the S3 bucket, we should delete our copy. If they need to have another copy (i.e. they deleted in an error), they can ask for it again.

**Comment by infinitewarp** [comment]
@jlalbertson for additional comment per our offline discussion, and then we may close this issue.

**Comment by jlalbertson** [comment]
@chargio  - I recommend we close this issue, replacing it with #1024. #1024 allows us to copy the **normalized** data to our S3 bucket - significantly less data than if we stored all of the raw customer data. Because the customer has access to their raw data already, it does not make sense for us to collect and store that for retrieval from our system. Plus, the cost of us collecting and storing all that data would become extremely high very quickly. 

Previously we had discussed having the data available for the customer to import into their own reporting tool. They still have that ability, exporting the data from their S3 bucket. While there are some use cases where we think it *might* be useful for the customer to get their data from us, the cost of storing that seems to be prohibitive. 

If in the future we find that there is a request by customers for them to retrieve their data from our S3 bucket, as opposed to their S3 bucket, we could open this topic up for discussion again.
